### PR TITLE
Gnome 45

### PR DESCRIPTION
**Pull request by philippun1**
Most of the work is done, the preferences dialog works again and if some of the remaining problematic code is commented out, the panel shows again.

The changes made with gnome 45 are not backwards compatible.

- [x] Migrate imports to ESM import
- [x] Export classes and functions
- [x] Implement ExtensionPreferences and Extension class
- [x] Fix remaining issues

### CODE REVIEW

File: extension.js [hunk 1]
```diff
@@ -18,54 +18,102 @@
...
-    extensionChangedHandler = extensionSystem.connect('extension-state-changed', (data, extension) => {
-        if (extension.uuid === UBUNTU_DOCK_UUID && extension.state === 1) {
-            _enable();
+        // Utils.removeKeybinding('open-application-menu');
+        // Utils.addKeybinding(
+        //     'open-application-menu',
+        //     new Gio.Settings({ schema_id: WindowManager.SHELL_KEYBINDINGS_SCHEMA }),
+        //     Main.wm._toggleAppMenu.bind(Main.wm),
+        //     Shell.ActionMode.NORMAL | Shell.ActionMode.POPUP
+        // );
```
Comment (philippun1) [inline extension.js:98]: open-application-menu and _toggleAppMenu are no longer available, do we have to replace this somehow?

File: extension.js [hunk 2]
```diff
@@ -88,63 +136,34 @@ function _enable() {
...
+    //     'open-application-menu',
+    //     new Gio.Settings({ schema_id: WindowManager.SHELL_KEYBINDINGS_SCHEMA }),
+    //     () => {
+    //         if(SETTINGS.get_boolean('show-appmenu'))
+    //             Main.wm._toggleAppMenu();
+    //         else
+    //             panelManager.primaryPanel.taskbar.popupFocusedAppSecondaryMenu();
+    //     },
+    //     Shell.ActionMode.NORMAL | Shell.ActionMode.POPUP
+    // );
```
Comment (philippun1) [inline extension.js:168]: open-application-menu and _toggleAppMenu are no longer available, do we have to replace this somehow?

**Review by philippun1 — COMMENTED** [review]
Everything seems to work again, I marked two code sections with code that has been removed. I am not sure if or with what they have to be replaced. The show apps button does not work, so I assume this is a direct consequence of this.

**Comment by darkblaze69** [comment]
I have this error. Am I missing smth?

`GLib.FileError: Failed to open file “/home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/schemas/gschemas.compiled”: open() failed: No such file or directory`

**Comment by philippun1** [comment]
> I have this error. Am I missing smth?
> 
> `GLib.FileError: Failed to open file “/home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/schemas/gschemas.compiled”: open() failed: No such file or directory`

You have to compile the schema. Open a terminal in the schema folder and run
```
glib-compile-schemas .
```

**Comment by tomking66** [comment]
Hi is your code working? I downloaded your version dash-to-panel for Gnome 45 and tested. it is still not working and the taskbar disappeared on Fedora 39 development version.

**Comment by tomking66** [comment]
I am getting this error message : "Proximity.Mode is undefined " and top panel disappeared. That makes Fedora 39  totally unstable with Gnome 45

**Comment by philippun1** [comment]
What gnome version does Fedora 39 have? I tested it directly with Gnome OS with Gnome 45.beta.1.

**Comment by darkblaze69** [comment]
For me it worked.

Having this error too ("Proximity.Mode is undefined") when enable:
Style > 'Override рапе! theme background opacity' and after that 'Dynamic background opacity'.

**Comment by philippun1** [comment]
Ok, I exported Proximity.Mode so autohide and dynamic background opacity works now.

File: prefs.js [hunk 3]
```diff
@@ -20,22 +20,22 @@
...
+import GLib from 'gi://GLib';
+import GObject from 'gi://GObject';
+import Gtk from 'gi://Gtk';
+import Adw from 'gi://Adw';
+import Gdk from 'gi://Gdk';
+
+import * as PanelSettings from './panelSettings.js';
+import * as Pos from './panelPositions.js';
+
+import {ExtensionPreferences, gettext as _} from 'resource:///org/gnome/Shell/Extensions/js/extensions/prefs.js';
```
Comment (AndrewZaech) [inline prefs.js:34]:
```suggestion
import {ExtensionPreferences, gettext as _, ngettext} from 'resource:///org/gnome/Shell/Extensions/js/extensions/prefs.js';
```

Also on line 687 change `Gettext.ngettext` to `ngettext`

File: prefs.js [hunk 4]
```diff
@@ -2081,7 +2082,7 @@ const Preferences = class {
 
         // About Panel
 
-        this._builder.get_object('extension_version').set_label(Me.metadata.version.toString() + (Me.metadata.commit ? ' (' + Me.metadata.commit + ')' : ''));
+        // TODO this._builder.get_object('extension_version').set_label(this._settings.version.toString() + (this._settings.commit ? ' (' + this._settings.commit + ')' : ''));
```
Comment (AndrewZaech) [inline prefs.js:2085]:
```suggestion
        const extension = ExtensionPreferences.lookupByURL(import.meta.url);
        const {metadata} = extension;
        this._builder.get_object('extension_version').set_label(metadata.version.toString() + (metadata.commit ? ' (' + metadata.commit + ')' : ''));
```

You can access the metadata like this

**Comment by AndrewZaech** [comment]
Thank you for working on the port! I'll try to help test it out.

I did notice issues when using multiple monitors and panels. From what I can see the issue is GNOME shell isn't exporting QuickSettings or ActivitiesButton from panel.js

**Comment by philippun1** [comment]
@AndrewZaech Thanks, I integrated your suggestions. I can also reproduce the multi monitor issue.

File: taskbar.js [hunk 5]
```diff
@@ -22,39 +22,40 @@
...
+import St from 'gi://St';
+
+import * as AppFavorites from 'resource:///org/gnome/shell/ui/appFavorites.js';
+import * as Dash from 'resource:///org/gnome/shell/ui/dash.js';
+import * as DND from 'resource:///org/gnome/shell/ui/dnd.js';
+import * as IconGrid from 'resource:///org/gnome/shell/ui/iconGrid.js';
+import * as Main from 'resource:///org/gnome/shell/ui/main.js';
+import * as PopupMenu from 'resource:///org/gnome/shell/ui/popupMenu.js';
+import * as Workspace from 'resource:///org/gnome/shell/ui/workspace.js';
+import * as SearchController from 'resource:///org/gnome/shell/ui/searchController.js';
```
Comment (AndrewZaech) [inline taskbar.js:42]:
```suggestion
```
Remove this import and instead use `const SearchController = Main.overview.searchController;` to resolve "Show Apps" button issues

Comment (philippun1) [reply taskbar.js]: Ah nice, now it works. Thanks.

**Comment by tomking66** [comment]
Hi Philliphun1, I have tested your latest source code. Taskbar  did not disappear now. I was able to change some configurations. I noticed some other issues 1) ArcMenu (porting Gnome 45) stopped working, though the ArcMenu  icon showing in the taskbar, clicking to see the pull-down menu does not work. I see this error message "Extension point conflict: there is already a status indicator for role ArcMenu". This extension Arcmenu seems to be working very well with "App Icon Task Bar" - a light-weight version similar to Dash to Panel with less configurations. Enabling dash to panel also kills some other extensions like "AppIndicator and KStatusNotifierItem Support" which already available for Gnome 45. I did not get time to test everything, but I am sure, Dash To Panel is breaking some other extensions which otherwise work  fine.

**Comment by tomking66** [comment]
Thanks for your efforts to make the Dash To panel working for Gnome 45. It is the most comprehensive extension that can make many small extensions irrelevant and I only use two  Extensions:  Dash To Panel and ArcMenu to have my Gnome Workflow feeling seamless. BTW many people like Dash To Dock which has a ported version for Gnome 45 and it is working fine with ArcMenu.

**Comment by AndrewZaech** [comment]
> I noticed some other issues 1) ArcMenu (porting Gnome 45) stopped working, though the ArcMenu icon showing in the taskbar, clicking to see the pull-down menu does not work. I see this error message "Extension point conflict: there is already a status indicator for role ArcMenu". 

That was an issue with ArcMenu. It should be fixed with https://gitlab.com/arcmenu/ArcMenu/-/commit/670d861c7d18e00ec2544fa28f9bfa4d94ba38f7. Please try re-installing from the gnome-45 branch to confirm. Thanks!

**Comment by philippun1** [comment]
I can confirm the latest ArcMenu gnome-45 branch works fine with this branch.

**Comment by tomking66** [comment]
Thanks - I have downloaded the latest version of ArcMenu  for Gnome 45 and it is working fine now now.

**Comment by alumni** [comment]
Works fine with my setup :)

**Comment by nater1983** [comment]
I am recieving this error

**Comment by darkblaze69** [comment]
It needs some treatment for the new Light Style Shell (official extension), especially in Overview.

Screenshot from 2023-08-25 13-35-16

File: panel.js [hunk 6]
```diff
@@ -1209,7 +1187,7 @@ var Panel = GObject.registerClass({
     }
 
     _onPanelMouseScroll(actor, event) {
-        let scrollAction = Me.settings.get_string('scroll-panel-action');
+        let scrollAction = SETTINGS.get_string('scroll-panel-action');
         let direction = Utils.getMouseScrollDirection(event);
 
         if (!this._checkIfIgnoredScrollSource(event.get_source()) && !this._timeoutsHandler.getId(T6)) {
```
Comment (AndrewZaech) [inline panel.js:1193]:
```suggestion
        const targetActor = global.stage.get_event_actor(event);

        if (!this._checkIfIgnoredScrollSource(targetActor) && !this._timeoutsHandler.getId(T6)) {
```
https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2216

This should fix https://github.com/home-sweet-gnome/dash-to-panel/pull/1939#issuecomment-1686780623

Comment (philippun1) [reply panel.js]: Thanks, that works.

**Comment by philippun1** [comment]
@nater1983 You are using version 56 which does not have the patches for gnome 45 yet.
@darkblaze69 I can confirm the behavior. But since the light style can only be enabled by an extension and is not default in gnome 45 this may go beyond the scope of this PR. I will take a look into it.

**Comment by darkblaze69** [comment]
> @darkblaze69 I can confirm the behavior. But since the light style can only be enabled by an extension and is not default in gnome 45 this may go beyond the scope of this PR. I will take a look into it.

I agree.

**Comment by nater1983** [comment]
[philippun1](https://github.com/philippun1) actually I patched it 
snippit from my slackbuild ```
git clone https://github.com/home-sweet-gnome/dash-to-panel.git
cd $SRCNAM
git checkout $_commit
patch -Np1 -i $CWD/1939.patch
```

**Comment by nater1983** [comment]
It works this time even with ArcMenu
image

**Comment by darkblaze69** [comment]
Cannot reorder app icons on the dash panel.

**Comment by philippun1** [comment]
@darkblaze69 Should work again.

**Comment by darkblaze69** [comment]
Today gnome-shell and mutter migrated to Mtk library and dash-to-panel stopped working. Is it now fixable by dash-to-panel or probably some problem on gnome-shell side?

<details><summary>Logs</summary>
<p>

```
Aug 31 15:13:14 gnome-shell[3171]: JS ERROR: Extension dash-to-panel@jderose9.github.com: TypeError: (intermediate value).Rectangle is not a constructor
                                   _updateWatchRect@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/proximity.js:59:21
                                   ProximityWatch@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/proximity.js:49:14
                                   createWatch@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/proximity.js:83:21
                                   _updateProximityWatch@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/transparency.js:132:61
                                   DynamicTransparency@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/transparency.js:45:14
                                   enable@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panel.js:214:36
                                   _createPanel@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:431:15
                                   enable@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/panelManager.js:76:38
                                   _enable@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/extension.js:147:18
                                   enable@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/extension.js:82:16
                                   _callExtensionEnable@resource:///org/gnome/shell/ui/extensionSystem.js:250:38
                                   async*_onEnabledExtensionsChanged@resource:///org/gnome/shell/ui/extensionSystem.js:599:24
                                   async*_onUserExtensionsEnabledChanged@resource:///org/gnome/shell/ui/extensionSystem.js:585:20
                                   _loadExtensions/<@resource:///org/gnome/shell/ui/extensionSystem.js:680:18
                                   createCheckedMethod/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:730:46
                                   set UserExtensionsEnabled@resource:///org/gnome/shell/ui/shellDBus.js:482:25
                                   _handlePropertySet@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:395:35
                                   _wrapJSObject/<@resource:///org/gnome/gjs/modules/core/overrides/Gio.js:414:35
                                   @resource:///org/gnome/shell/ui/init.js:21:20
Aug 31 15:13:14 gnome-shell[3171]: JS ERROR: TypeError: (intermediate value).Rectangle is not a constructor
                                   _updateWatchRect@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/proximity.js:59:21
                                   ProximityWatch/this._allocationChangedId<@file:///home/user/.local/share/gnome-shell/extensions/dash-to-panel@jderose9.github.com/proximity.js:47:84
                                   @resource:///org/gnome/shell/ui/init.js:21:20
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <panelLeft>[<StBoxLayout>:0x56142f5a2220] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <panelCenter>[<StBoxLayout>:0x56142f5a3080] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <panelRight>[<StBoxLayout>:0x56142f5adc50] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f85a660] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f860ed0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f8671a0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f8b2ea0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f9416d0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_status_keyboard_InputSourceIndicator>:0x56142f93cac0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_status_keyboard_InputSourceIndicatorContainer>:0x56142f943420] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StLabel>:0x56142f95f5e0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterText>:0x56142f960120] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f5b1060] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <panelActivities>[<Gjs_ui_panel_ActivitiesButton>:0x56142f5aff50] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_panel_WorkspaceIndicators>:0x56142f5b2860] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_panel_WorkspaceDot>:0x56142f5b5560] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StWidget>:0x56142f5b6130] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_panel_WorkspaceDot>:0x56143134bf80] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StWidget>:0x56143134c6b0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142e2f9040] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_panel_QuickSettings>:0x56142e2fabf0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBoxLayout>:0x56142e2ee9f0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_status_network_Indicator>:0x56143057e8e0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StIcon>:0x561430df90c0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterActor>:0x56142deb3250] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StIcon>:0x561430dfa100] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterActor>:0x561432405b60] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_status_volume_OutputIndicator>:0x561430ebbea0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StIcon>:0x561430ebcec0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterActor>:0x56143213a380] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_status_system_Indicator>:0x5614304ce020] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StIcon>:0x5614304b5d60] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterActor>:0x5614300a1a60] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x56142f5b8f90] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_dateMenu_DateMenuButton>:0x56142f5b81a0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBoxLayout>:0x56142f5c4380] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StLabel>:0x56142f5bf310] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterText>:0x56142f5bfdb0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_dash_ShowAppsIcon>:0x5614325349c0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StButton>:0x5614313fe3e0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<Gjs_ui_iconGrid_BaseIcon>:0x561432160580] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBoxLayout>:0x561431420dc0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StBin>:0x5614325008b0] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<StIcon>:0x561431400590] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: Can't update stage views actor <unnamed>[<ClutterActor>:0x56142e76a060] is on because it needs an allocation.
Aug 31 15:13:14 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:14 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:14 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:14 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:15 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:16 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:17 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:18 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:19 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:20 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:21 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:22 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:23 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:24 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:25 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed
Aug 31 15:13:26 gnome-shell[3171]: cogl_framebuffer_set_viewport: assertion 'width > 0 && height > 0' failed

```

</p>
</details>

**Comment by darkblaze69** [comment]
> Today gnome-shell and mutter migrated to Mtk library and dash-to-panel stopped working.

should be fixed in https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2930

**Comment by crosser** [comment]
Confirm working (at least did not notice troubles so far) with gnome 45.beta that was just pushed to ubuntu mantic.

**Comment by git-Hmmm** [comment]
image
volume action is broken

**Comment by darkblaze69** [comment]
After this https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2936 
(st/types: Remove StAlign) it stopped working.

error: `(intermediate value).Align is undefined`

**Comment by philippun1** [comment]
@git-Hmmm Volume scroll fixed.
@darkblaze69 Align fixed.

**Comment by philippun1** [comment]
Gnome 45.rc is released and I did some final fixes. In my opinion this PR is ready!
@charlesg99 What do you think?

**Comment by git-Hmmm** [comment]
@philippun1 this bar animation on top
doesn't work

it used to be animated previously
like when minimize, expand window
it used to collpase nicely 

image

now it just collapses instantly 
no animation in bar.

**Comment by rathann** [comment]
@unit-404 I've just built an update with this PR applied for [rawhide](https://bodhi.fedoraproject.org/updates/FEDORA-2023-d755e3316e) and made it part of the GNOME 45 [mega-update](https://bodhi.fedoraproject.org/updates/FEDORA-2023-d422824191) for F39, too. Works fine for me.

**Comment by philippun1** [comment]
@unit-404 you are probably not using this branch.
@git-Hmmm What animation do you mean exactly?

**Comment by charlesg99** [comment]
@philippun1 Awesome, thanks! I'll test it this week!

**Comment by zofoss-team** [comment]
@philippun1 
image
see that green bar

when you click on it, it collapses and minimize the application

the bar used to collapse:
- previously with animation
- now without animation

**Comment by git-Hmmm** [comment]
@philippun1 
image
see that green bar

when you click on it, it collapses and minimize the application

the bar used to collapse:
- previously with animation
- now without animation

**Comment by charlesg99** [comment]
Thanks Philipp!

**Comment by git-Hmmm** [comment]
@philippun1 I installed the release version and the issue i reported still isn't addressed
this one: https://github.com/home-sweet-gnome/dash-to-panel/pull/1939#issuecomment-1712945957

**Pull request merged**